File: Python/Arrays-and-hashing/Contains-duplicate/contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        """
        Given an integer array nums, return true if any value appears at least twice in the array, 
        and return false if every element is distinct.
        """
        hashset = set()
        for x in nums:            
            if x in hashset:
                return True
            hashset.add(x)
        return False

File: Ruby/lib/arrays_and_hashing/contains_duplicate.rb
# frozen_string_literal: true
class Solution
  # @param {Integer[]} nums
  # @return {Boolean}
  def contains_duplicate(nums)
    hash = {}
    nums.each do |num|
      if hash.has_key? num
        return true
      end
      hash[num] = true
    end

    false
  end
end
